api/src/infrastructure/kubernetes/infrastructure.rs

```diff
@@ -478,6 +495,136 @@ impl KubernetesInfrastructure {
+                            app_name,
+                            service.service_name()
+                        )),
+                        labels: Some(BTreeMap::from([
+                            (APP_NAME_LABEL.to_owned(), app_name.to_owned()),
+                            (
+                                SERVICE_NAME_LABEL.to_owned(),
+                                service.service_name().to_owned(),
+                            ),
+                            (
+                                STORAGE_TYPE_LABEL.to_owned(),
+                                declared_volume
+                                    .split('/')
+                                    .last()
+                                    .unwrap_or("default")
+                                    .to_owned(),
+                            ),
+                        ])),
+                        ..Default::default()
+                    },
+                    spec: Some(PersistentVolumeClaimSpec {
+                        storage_class_name: storage_class.to_owned(),
+                        access_modes: Some(vec!["ReadWriteOnce".to_owned()]),
+                        resources: Some(ResourceRequirements {
+                            requests: Some(BTreeMap::from_iter(vec![(
+                                "storage".to_owned(),
+                                Quantity(
+                                    storage_size.to_owned().unwrap_or_else(|| "2Gi".to_owned()),
+                                ),
+                            )])),
+                            ..Default::default()
+                        }),
+                        ..Default::default()
+                    }),
+                    ..Default::default()
+                };
+
+                match Api::namespaced(client.clone(), app_name)
+                    .create(&PostParams::default(), &persistent_volume_claim)
+                    .await
+                {
+                    Ok(pvc) => {
+                        persistent_volume_map.insert(declared_volume, pvc);
+                    }
+                    Err(e) => {
+                        error!("Cannot deploy persistent volume claim: {}", e);
+                        return Ok(persistent_volume_map);
+                    }
+                }
+            } else {
+                if fetched_pvc.len() != 1 {
+                    warn!(
+                        "Found more than 1 Persistent Volume Claim - {:?} for declared image path {} \n Using the first available Persistent Volume Claim - {:?}",
+                        &fetched_pvc,
+                        declared_volume,
+                        &fetched_pvc[0].metadata.name
+                    );
+                }
+                persistent_volume_map.insert(declared_volume, fetched_pvc[0].clone());
+            }
+        }
+        Ok(persistent_volume_map)
+    }
+
+    async fn fetch_default_storage_class(&self) -> Option<String> {
```

Comment:
Why string and not `StorageClass`?

```suggestion
    async fn fetch_default_storage_class(&self) -> Option<StorageClass> {
```